kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp

```diff
@@ -0,0 +1,220 @@
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex()
+    {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::LevelSetConvectionElementSimplexAlgebraicStabilization(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex(NewId, pGeometry)
+    {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::LevelSetConvectionElementSimplexAlgebraicStabilization(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+    : LevelSetConvectionElementSimplex<TDim, TNumNodes>::LevelSetConvectionElementSimplex(NewId, pGeometry, pProperties)
+    {}
+
+    /// Destructor.
+    template< unsigned int TDim, unsigned int TNumNodes>
+    LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::~LevelSetConvectionElementSimplexAlgebraicStabilization() {}
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    Element::Pointer LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const
+    {
+        KRATOS_TRY
+        return Element::Pointer(new LevelSetConvectionElementSimplexAlgebraicStabilization(NewId, this->GetGeometry().Create(ThisNodes), pProperties));
+        KRATOS_CATCH("");
+    }
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    void LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::CalculateLocalSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        if (rLeftHandSideMatrix.size1() != TNumNodes)
+            rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false); //false says not to preserve existing storage!!
+
+        if (rRightHandSideVector.size() != TNumNodes)
+            rRightHandSideVector.resize(TNumNodes, false); //false says not to preserve existing storage!!
+
+        const double delta_t = rCurrentProcessInfo[DELTA_TIME];
+        const double dt_inv = 1.0 / delta_t;
+        const double theta = rCurrentProcessInfo.Has(TIME_INTEGRATION_THETA) ? rCurrentProcessInfo[TIME_INTEGRATION_THETA] : 0.5;
+
+        auto p_conv_diff_settings = rCurrentProcessInfo.GetValue(CONVECTION_DIFFUSION_SETTINGS);
+        const auto& rUnknownVar = p_conv_diff_settings->GetUnknownVariable();
+        const auto& rConvVar = p_conv_diff_settings->GetConvectionVariable();
+        const auto& rGradVar = p_conv_diff_settings->GetGradientVariable();
```

Comment:
```suggestion
        const auto& r_unknown_var = p_conv_diff_settings->GetUnknownVariable();
        const auto& r_conv_var = p_conv_diff_settings->GetConvectionVariable();
        const auto& r_grad_var = p_conv_diff_settings->GetGradientVariable();
```
Snake case for local variables (style guide convention).